### PR TITLE
Analysis refactor GUI part1

### PR DESCRIPTION
**Pull request by josenavas**
The file `qiita_pet/handlers/analysis_handlers/listing_handlers.py` contains handlers that already existed before, I just moved them to create a better structure of the code base so it is easier to maintain. They may change a bit in a later PR.

The main changes are done in the file `qiita_pet/handlers/analysis_handlers/base_handlers.py`.

This gif shows how the new analysis description page looks like.

new_analysis_desc_page

The artifact page is not loading ATM because there are some changes that I need to do on that handler - however this PR is already big enough that I refer those changes for a future PR.

@antgonza this is ready for review.

### CODE REVIEW

File: qiita_pet/handlers/analysis_handlers/base_handlers.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+from qiita_pet.handlers.util import to_int
+from qiita_db.analysis import Analysis
+
+
+class CreateAnalysisHandler(BaseHandler):
+    @authenticated
+    @execute_as_transaction
+    def post(self):
+        name = self.get_argument('name')
+        desc = self.get_argument('description')
```
Comment (antgonza) [inline qiita_pet/handlers/analysis_handlers/base_handlers.py]: Does this make a desc required or do we need to put a default value?

File: qiita_pet/handlers/analysis_handlers/base_handlers.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        # Loop through all the nodes in artifact descendants graph
+        for n in g.nodes():
+            # Get if the object is an artifact or a job
+            obj_type = n[0]
+            # Get the actual object
+            obj = n[1]
+            if obj_type == 'job':
+                name = obj.command.name
+            else:
+                if full_access or obj.visibility == 'public':
```
Comment (antgonza) [inline qiita_pet/handlers/analysis_handlers/base_handlers.py]: This if confused me for how it's written, perhaps better:
```python
if not full_access and not obj.visibility == 'public':
    continue
```

File: qiita_pet/handlers/analysis_handlers/base_handlers.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                    name = '%s - %s' % (obj.name, obj.artifact_type)
+                else:
+                    continue
+            nodes.add((obj_type, obj.id, name))
+
+        edges.update({(s[1].id, t[1].id) for s, t in g.edges()})
+
+    # Transforming to lists so they are JSON serializable
```
Comment (antgonza) [inline qiita_pet/handlers/analysis_handlers/base_handlers.py]: Can you improve this documentation?

File: qiita_pet/handlers/analysis_handlers/listing_handlers.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        check_analysis_access(self.current_user, analysis)
+
+        try:
+            Analysis.delete(analysis_id)
+            msg = ("Analysis <b><i>%s</i></b> has been deleted." % (
+                analysis_name))
+            level = "success"
+        except Exception as e:
+            e = str(e)
+            msg = ("Couldn't remove <b><i>%s</i></b> analysis: %s" % (
```
Comment (antgonza) [inline qiita_pet/handlers/analysis_handlers/listing_handlers.py]: Is it possible to add a reason of why? If too difficult, that's fine.

File: qiita_pet/webserver.py [hunk 5]
```diff
@@ -90,18 +89,20 @@ def __init__(self):
             (r"/profile/", UserProfileHandler),
             (r"/user/messages/", UserMessagesHander),
             (r"/user/jobs/", UserJobs),
-            (r"/results/(.*)", ResultsHandler,
-             {"path": RES_PATH}),
+            # (r"/results/(.*)", ResultsHandler,
```
Comment (antgonza) [inline qiita_pet/webserver.py]: worth removing?

File: qiita_pet/webserver.py [hunk 6]
```diff
@@ -90,18 +89,20 @@ def __init__(self):
...
+            #  {"path": RES_PATH}),
             (r"/static/(.*)", tornado.web.StaticFileHandler,
              {"path": STATIC_PATH}),
-            (r"/analysis/3", SelectCommandsHandler),
-            (r"/analysis/wait/(.*)", AnalysisWaitHandler),
-            (r"/analysis/results/(.*)", AnalysisResultsHandler),
-            (r"/analysis/show/", ShowAnalysesHandler),
+            # (r"/analysis/wait/(.*)", AnalysisWaitHandler),
```
Comment (antgonza) [inline qiita_pet/webserver.py]: Also this?

File: qiita_pet/handlers/analysis_handlers/base_handlers.py (hunk 1 above)
Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers/base_handlers.py]: No it's not required (same behavior as before). If the user doesn't introduce anything, the description will be blank (empty string).

File: qiita_pet/handlers/analysis_handlers/base_handlers.py (hunk 2 above)
Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers/base_handlers.py]: Sure - I've also added a comment so it is more clear.

File: qiita_pet/handlers/analysis_handlers/base_handlers.py (hunk 3 above)
Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers/base_handlers.py]: Sure

File: qiita_pet/handlers/analysis_handlers/listing_handlers.py (hunk 4 above)
Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers/listing_handlers.py]: The last part of the message includes the `str(e)` so the reason should be in there.

File: qiita_pet/webserver.py (hunk 5 above)
Comment (josenavas) [inline qiita_pet/webserver.py]: Done

File: qiita_pet/webserver.py (hunk 6 above)
Comment (josenavas) [inline qiita_pet/webserver.py]: Done

**Pull request merged**